lib/std/os/linux.zig

```diff
@@ -1227,3 +1227,464 @@ test "" {
+        clone = 1 << 9,
+        /// Enable memory-mapping BPF map
+        mmapable = 1 << 10,
+    };
+
+    pub const ProgQueryFlags = enum(64) {
+        /// Query effective (directly attached + inherited from ancestor
+        /// cgroups) programs that will be executed for events within a cgroup.
+        /// attach_flags with this flag are returned only for directly attached
+        /// programs.
+        effective = 1,
+    };
+
+    pub const MapType = enum(u32) {
+        unspec,
+        hash,
+        array,
+        prog_array,
+        perf_event_array,
+        percpu_hash,
+        percpu_array,
+        stack_trace,
+        cgroup_array,
+        lru_hash,
+        lru_percpu_hash,
+        lpm_trie,
+        array_of_maps,
+        hash_of_maps,
+        devmap,
+        sockmap,
+        cpumap,
+        xskmap,
+        sockhash,
+        cgroup_storage,
+        reuseport_sockarray,
+        percpu_cgroup_storage,
+        queue,
+        stack,
+        sk_storage,
+        devmap_hash,
+        struct_ops,
+        ringbuf,
+    };
+
+    pub const AttachType = enum(u32) {
+        cgroup_inet_ingress,
+        cgroup_inet_egress,
+        cgroup_inet_sock_create,
+        cgroup_sock_ops,
+        sk_skb_stream_parser,
+        sk_skb_stream_verdict,
+        cgroup_device,
+        sk_msg_verdict,
+        cgroup_inet4_bind,
+        cgroup_inet6_bind,
+        cgroup_inet4_connect,
+        cgroup_inet6_connect,
+        cgroup_inet4_post_bind,
+        cgroup_inet6_post_bind,
+        cgroup_udp4_sendmsg,
+        cgroup_udp6_sendmsg,
+        lirc_mode2,
+        flow_dissector,
+        cgroup_sysctl,
+        cgroup_udp4_recvmsg,
+        cgroup_udp6_recvmsg,
+        cgroup_getsockopt,
+        cgroup_setsockopt,
+        trace_raw_tp,
+        trace_fentry,
+        trace_fexit,
+        modify_return,
+        lsm_mac,
+        trace_iter,
+        cgroup_inet4_getpeername,
+        cgroup_inet6_getpeername,
+        cgroup_inet4_getsockname,
+        cgroup_inet6_getsockname,
+        xdp_devmap,
+    };
+
+    pub const Cmd = enum(usize) {
+        map_create,
+        map_lookup_elem,
+        map_update_elem,
+        map_delete_elem,
+        map_get_next_key,
+        prog_load,
+        obj_pin,
+        obj_get,
+        prog_attach,
+        prog_detach,
+        prog_test_run,
+        prog_get_next_id,
+        map_get_next_id,
+        prog_get_fd_by_id,
+        map_get_fd_by_id,
+        obj_get_info_by_fd,
+        prog_query,
+        raw_tracepoint_open,
+        btf_load,
+        btf_get_fd_by_id,
+        task_fd_query,
+        map_lookup_and_delete_elem,
+        map_freeze,
+        btf_get_next_id,
+        map_lookup_batch,
+        map_lookup_and_delete_batch,
+        map_update_batch,
+        map_delete_batch,
+        link_create,
+        link_update,
+        link_get_fd_by_id,
+        link_get_next_id,
+        enable_stats,
+        iter_create,
```

Comment:
Should this be a non-exhaustive enum?

Reply:
What do you mean by exhaustive enum? simply writing each field as its own variable? I realize from reading os/bits/linux.zig that we keep naming/types close to the headers.

Reply:
https://ziglang.org/documentation/master/#Non-exhaustive-enum

You use them when e.g. the kernel might add new values to the enum in future.

Reply:
oh hot damn